### PR TITLE
Factories sophia – tests

### PR DESCRIPTION
**Pull request by radrow**
This PR fixes the test contracts to use the new factory syntax. It also adds tests for `CLONE`, `CREATE` and `BYTECODE_HASH`.

### CODE REVIEW

File: apps/aecontract/test/aecontract_SUITE.erl [hunk 1]
```diff
@@ -5560,31 +5573,32 @@ sophia_protected_call(_Cfg) ->
                 {Fun, MinGas, MaxGas, Res, NewState - OldState, [ClientBal1 - ClientBal0, ProxyBal1 - ProxyBal0, ServerBal1 - ServerBal0]}
            end,
     {test_ok, _, _, {116, _}, _, _} = Test(test_ok, word, 0, 0),
-    Results = [ Test(test_wrong_ret,     {option, bool}, 150, 200)
```
Comment (radrow) [inline apps/aecontract/test/aecontract_SUITE.erl:5563]: This worries me, but somehow I needed to fix these

**Comment by genievot** [comment]
Hi @radrow 
Can you please point me to CREATE tests, I want to use them for my project.

**Comment by genievot** [comment]
Hi @radrow
Can you please point me to CREATE tests, I would like to use them for my project.

**Comment by genievot** [comment]
Hi @radrow
Thank you for adding those tests.

**Comment by hanssv** [comment]
This needs a rebase and also would be good if the tests used realistic gas amounts (< 6M).

File: rebar.lock [hunk 2]
```diff
@@ -1,190 +0,0 @@
-{"1.2.0",
```
Comment (cytadela8) [inline rebar.lock:1]: ?

File: test/contracts/test.aes [hunk 3]
```diff
@@ -84,17 +84,17 @@ contract Identity =
...
   // | (x::ys) => f(x)::map(f,ys)
   // }
   // let id(x) = x
-  // let main(xs) = map(double,xs)
+  // let main_(xs) = map(double,xs)
   entrypoint z(f,x) = x
   function s(n) = (f,x)=>f(n(f,x))
   function add(m,n) = (f,x)=>m(f,n(f,x))
-  entrypoint main(_) =
+  entrypoint main_(_) =
```
Comment (cytadela8) [inline test/contracts/test.aes:97]: What's up with `main` is this some new keyword?

Comment (radrow) [reply test/contracts/test.aes:97]: yes

File: rebar.lock (hunk 2 above)
Comment (radrow) [reply rebar.lock:1]: wtf

**Review by cytadela8 — APPROVED** [review]

**Review by hanssv — APPROVED** [review]
Note: I reverted the changes to aecontract_SUITE:sophia_protected_call - they were indeed worrying, had they'd been necessary...

LGTM

**Review by UlfNorell — APPROVED** [review]

**Pull request merged**